server/implementation/src/main/java/io/smallrye/graphql/execution/ExecutionService.java

```diff
@@ -121,7 +124,7 @@ public ExecutionResponse execute(JsonObject jsonInput) {
                 eventEmitter.fireAfterExecute(context);
 
                 ExecutionResponse executionResponse = new ExecutionResponse(executionResult, config);
-                if (config.logPayload()) {
+                if (config.logPayload().equals(Config.logPayloadOption.TRUE)) {
```

Comment:
I would move all this (logging of payload) to a method that can just be called from both places.